File: python/xor/xorstarter.py
#!/usr/bin/env python3

import sys
from crypto.Util.number import *

if sys.version_info.major == 2:
    print("You are running Python 2, which is no longer supported. Please update to Python 3.")

label = input("Please enter the integer: ")
key   = int(input("please enter the key: "))
flag = "crypto{"
for c in label:
    val  = ord(c) ^ key
    flag = flag + str(chr(val))
flag = flag + "}"; 
#print the flag
print(flag)

File: c++/xor/xorstarter.cpp
#include <iostream>
#include <string>

int main() {
    std::string label;
    int key;
    std::string flag = "crypto{";

    std::cout << "Please enter the integer: ";
    std::getline(std::cin, label);
    std::cout << "please enter the key: ";
    std::cin >> key;

    for (char c : label) {
        int val = static_cast<int>(c) ^ key;
        flag += static_cast<char>(val);
    }
    flag += "}";

    // print the flag
    std::cout << flag << std::endl;

    return 0;
}
